config/src/main/kotlin/org/springframework/security/config/annotation/web/HttpSecurityDsl.kt

```diff
@@ -76,6 +78,33 @@ class HttpSecurityDsl(private val http: HttpSecurity, private val init: HttpSecu
 
     var authenticationManager: AuthenticationManager? = null
 
+    /**
+     * Applies a [SecurityConfigurerAdapter] to this [HttpSecurity]
+     *
+     * Example:
+     *
+     * ```
+     * @Configuration
+     * @EnableWebSecurity
+     * class SecurityConfig {
+     *
+     *     @Bean
+     *     fun securityFilterChain(http: HttpSecurity): SecurityFilterChain {
+     *         http {
+     *             apply(CustomSecurityConfigurer<HttpSecurity>())
+     *         }
+     *         return http.build()
+     *     }
+     * }
+     * ```
+     *
+     * @param configurer
+     * the [SecurityConfigurerAdapter] for further customizations
+     */
+    fun <C : SecurityConfigurerAdapter<DefaultSecurityFilterChain, HttpSecurity>> apply(configurer: C): C {
+        return this.http.apply(configurer)
```

Comment:
I think we can go one step further with this, and call the Kotlin `apply` method on a custom DSL block, like this:

```kotlin
fun <C : SecurityConfigurerAdapter<DefaultSecurityFilterChain, HttpSecurity>> apply(configurer: C, configuration: C.() -> Unit = { }): C {
    return this.http.apply(configurer).apply(configuration)
}
```

Then it could be used like this:

```kotlin
apply(CustomSecurityConfigurer<HttpSecurity>()) {
    customProperty = "..."
}
```

What do you think?

Reply:
that's cool, let me do some code changes